doc/adrs/095_simplify_the_manipulation_of_nodes_by_switching_to_a_common_box_model.adoc

```diff
@@ -0,0 +1,458 @@
+
+As a result, with a center label a node cannot have any child nodes.
+In this case, the inside label box would replace entirely the content box.
+
+
+=== Border box
+
+The border box is placed around the inside label box.
+Its size is determined by the requested size of the border.
+Technically each size of the border could have a different size like in CSS with `border-width: 0px 1px 2px 3px`.
+
+
+=== Border nodes box
+
+Border nodes will be placed in the border nodes box.
+It will not count torward the size of the node.
+If a specifier requires a node to have a size of 100x100, then the border nodes would still stick outside of this area.
+
+WARNING: They will not overlap the content box, inside label box nor the border box.
+
+While resizing a node, the border nodes will not be resized or modified in any way.
+They would just be moved to stay at the same location proportionnaly.
+For example, if a border node is located on the left of a node at 40% from the top, resizing the node would still keep the border node at 40% from the top if possible.
+
+The border nodes box cannot be modified by a request from a minimal footprint.
+
+
+=== Outside label box
+
+It does not count in the size of the node.
+It cannot be wrapped in the same way that border nodes cannot be impacted by a request for a smaller node footprint.
+
+WARNING: Strictly outside of the border nodes box, if we have border nodes on the bottom, the label is pushed further
+
+=== Margin box
+
+It is used to force nodes not be stuck one next to the others.
+
+
+
+
+=== Resizing
+
+The specifier may give us a prefered size for a node and we will try to use it as the current size of the node.
+The end user may also try to resize a node to their prefered size.
+We will also try to fulfill this request.
+
+==== Free form layout strategy
+
+Regardless of the request, resizing a node cannot change in any way the footprint of its children.
+The children of a node are a black box with regards to the parent.
+Resizing a node in this case will keep the border node proportionaly at their location if possible.
+
+
+==== List layout strategy
+
+
+- Content box and free form layout strategy
+- Content box and list layout strategy
+- min size horizontally?
+- min size vertically?
+- both?
+
+
+We will not wrap the labels of a child to let a parent be resized.
+
+
+==== Wrapped labels
+
+If the size of the node is not constrained (by its parent or by a size given by the specifier / the end user) then its label cannot be wrapped.
+The node will thus consider the place necessary to display the label.
+
+Labels can only be wrapped in their own inside label box.
+A wrapped label will thus increase the height of its inside label box.
+
+We will have to be able to compute the minimal size of the node using a wrapped label and not only using a non-wrapped label.
+
+
+
+== Status
+
+Work in progress
+
+
+== Consequences
+
```

Comment:
The ELK configuration will not be used anymore by this new box model, right? If it is the case, could you add it to this ADR please?